dev-packages/application-manager/src/generator/frontend-generator.ts

```diff
@@ -183,19 +183,36 @@ if (isMaster) {
                 shell.openExternal(url);
             });
 
-            const saveState = () => {
-                const bounds = newWindow.getBounds();
-                electronStore.set(windowStateName, {
-                    isMaximized: newWindow.isMaximized(),
-                    width: bounds.width,
-                    height: bounds.height,
-                    x: bounds.x,
-                    y: bounds.y
-                })
-            }
-            newWindow.on('close', saveState);
-            newWindow.on('resize', saveState);
-            newWindow.on('move', saveState);
+            // Save the window geometry state on every change
+            const saveWindowState = () => {
+                try {
+                    let bounds;
+                    if (newWindow.isMaximized()) {
+                        bounds = electronStore.get(WINDOW_STATE, {});
```

Comment:
What is the point of re-doing `electronStore.get(WINDOW_STATE, {})` when the window is maximized? Also:

- Couldn't you just re-use `windowState` from above?

- It seems your are calling `electronStore.set()` with the same values you got from `electronStore.get()`. Maybe just skip saving window state when it's maximized?

Reply:
> What is the point of re-doing `electronStore.get(WINDOW_STATE, {})` when the window is maximized?

It makes sure that the bounds are kept, so after unmaximizing we get the same bounds we had before maximizing.

> Couldn't you just re-use windowState from above?

No, the stored bounds might have been changed in the meantime. We could write the bounds into the local variable on every store operation if we want to avoid the additional `get`.

> It seems your are calling electronStore.set() with the same values you got from electronStore.get().

No, the `isMaximized` property is changed.